net/service.go

```diff
@@ -0,0 +1,30 @@
+package net
+
+import (
+	"context"
+	"errors"
+	"github.com/textileio/textile-go/pb"
+	inet "gx/ipfs/QmXfkENeeBvh3zYA51MaSdGUdBjhQ99cP5WQe8zgr6wchG/go-libp2p-net"
+	"gx/ipfs/QmZoWKhxUmZ2seW4BzX6fJkNR8hh9PsGModr7q171yq2SS/go-libp2p-peer"
+)
+
+var (
+	DuplicateMessage = errors.New("duplicate message")
+)
+
+type NetworkService interface {
```

Comment:
this is the highest level abstraction for the network service. send and handle messages from other peers